contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -0,0 +1,551 @@
+    if (s_subscriptions[subscriptionId].owner == address(0)) {
+      revert InvalidSubscription();
+    }
+  }
+
+  function _isValidConsumer(address client, uint64 subscriptionId) internal view {
+    if (s_consumers[client][subscriptionId].allowed == false) {
+      revert InvalidConsumer(subscriptionId, client);
+    }
+  }
+
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function getSubscription(
+    uint64 subscriptionId
+  )
+    external
+    view
+    returns (uint96 balance, uint96 blockedBalance, address owner, address requestedOwner, address[] memory consumers)
+  {
+    _isValidSubscription(subscriptionId);
+    return (
+      s_subscriptions[subscriptionId].balance,
+      s_subscriptions[subscriptionId].blockedBalance,
+      s_subscriptions[subscriptionId].owner,
+      s_subscriptions[subscriptionId].requestedOwner,
+      s_subscriptions[subscriptionId].consumers
+    );
+  }
+
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function getConsumer(
+    address client,
+    uint64 subscriptionId
+  ) external view returns (bool allowed, uint64 initiatedRequests, uint64 completedRequests) {
+    return (
+      s_consumers[client][subscriptionId].allowed,
+      s_consumers[client][subscriptionId].initiatedRequests,
+      s_consumers[client][subscriptionId].completedRequests
+    );
+  }
+
+  // ================================================================
+  // |                 Internal Payment methods                     |
+  // ================================================================
+  /**
+   * @notice Blocks funds on a subscription account to be used for an in flight request
+   * @dev Only callable within the Router
+   * @param client - the consumer contract that initiated the request
+   * @param subscriptionId - The subscription ID to block funds for
+   * @param amount - The amount to transfer
+   */
+  function _blockBalance(address client, uint64 subscriptionId, uint96 amount) internal {
+    s_subscriptions[subscriptionId].blockedBalance += amount;
+    s_consumers[client][subscriptionId].initiatedRequests += 1;
+  }
+
+  /**
+   * @notice Ensure that the subscription balance can still afford fulfillment cost
+   */
+  function _checkBalance(
+    uint96 estimatedCost,
+    uint256 gasAfterPaymentCalculation,
+    uint96 adminFee,
+    uint32 callbackGasLimit,
+    uint96 juelsPerGas,
+    uint96 costWithoutFulfillment
+  ) internal returns (uint8) {
+    // Check that the transmitter has supplied enough gas for the callback to succeed
+    if (gasleft() < callbackGasLimit + gasAfterPaymentCalculation) {
+      return 3; // IFunctionsRouter.FulfillResult.INSUFFICIENT_GAS;
+    }
+
+    // Check that the cost has not exceeded the quoted cost
+    if (adminFee + costWithoutFulfillment + (juelsPerGas * SafeCast.toUint96(callbackGasLimit)) > estimatedCost) {
+      return 4; // IFunctionsRouter.FulfillResult.INSUFFICIENT_SUBSCRIPTION_BALANCE
```

Comment:
`INSUFFICIENT_SUBSCRIPTION_BALANCE` isn't actually descriptive of the error that is occuring here.  The problem is not that the subscription balance was exceeded, but that the estimated cost was exceeded.